pkg/bootstrapper/bootstrapper.go

```diff
@@ -499,63 +502,99 @@ func (wmcb *winNodeBootstrapper) createKubeletService() error {
+	}
+	return nil
+}
+
+// createKubeletService creates a new kubelet service to our specifications
+func (wmcb *winNodeBootstrapper) createKubeletService(c mgr.Config, kubeletArgs []string) error {
 	ksvc, err := wmcb.svcMgr.CreateService(KubeletServiceName, filepath.Join(wmcb.installDir, "kubelet.exe"), c, kubeletArgs...)
 	if err != nil {
 		return err
 	}
 
-	var dependents []*mgr.Service
-	if dependentSvc, err := wmcb.svcMgr.OpenService(kubeletDependentSvc); err == nil {
-		dependents = append(dependents, dependentSvc)
-	}
-	wmcb.kubeletSVC, err = newKubeletService(ksvc, dependents)
+	wmcb.kubeletSVC, err = newKubeletService(ksvc, nil)
 	if err != nil {
 		return fmt.Errorf("could not initialize struct kubeletService: %v", err)
 	}
+	return nil
+}
 
-	if err := wmcb.kubeletSVC.setRecoveryActions(); err != nil {
-		return fmt.Errorf("failed to set recovery actions for Windows service %s", KubeletServiceName)
+// updateKubeletService updates an existing kubelet service with our specifications
+func (wmcb *winNodeBootstrapper) updateKubeletService(config mgr.Config, kubeletArgs []string) error {
+	// Get existing config
+	existingConfig, err := wmcb.kubeletSVC.config()
+	if err != nil {
+		return fmt.Errorf("no existing config found")
 	}
 
-	return nil
-}
+	// Stop the kubelet service as there could be open file handles from kubelet.exe on the plugin files
+	if err := wmcb.kubeletSVC.stop(); err != nil {
+		return fmt.Errorf("unable to stop kubelet service: %v", err)
+	}
+	// Populate existing config with non default values from desired config.
+	existingConfig.Dependencies = config.Dependencies
+	existingConfig.DisplayName = config.DisplayName
+	existingConfig.StartType = config.StartType
 
-// refreshServiceManager will disconnect and reconnect from the Windows service API. In order to complete certain
-// operations, there must be zero handlers to the API present on the system.
-func (wmcb *winNodeBootstrapper) refreshServiceManager() error {
-	var err error
-	if err = wmcb.Disconnect(); err != nil {
-		return err
+	// Create kubelet command to populate config.BinaryPathName
+	// Add a space after kubelet.exe followed by the stand alone args
+	kubeletcmd := " " + filepath.Join(wmcb.installDir, "kubelet.exe") + " "
+	var standaloneArgs string
+	var keyvalueArgs string
+	// Add rest of the args
+	for _, args := range kubeletArgs {
+		if strings.Contains(args, "=") {
+			keyvalueArgs += args + " "
+		} else {
+			standaloneArgs += args + " "
+		}
 	}
-	// We need to give Windows time to clean up the services we've marked for deletion
-	time.Sleep(serviceWaitTime)
-	wmcb.svcMgr, err = mgr.Connect()
-	return err
+	existingConfig.BinaryPathName = strings.TrimSpace(kubeletcmd + standaloneArgs + keyvalueArgs)
+
+	// Update service config and restart
+	if err := wmcb.kubeletSVC.refresh(existingConfig); err != nil {
+		return fmt.Errorf("unable to refresh kubelet service: %v", err)
+	}
```

Comment:
This shouldnt be necessary

Reply:
Wouldn`t we need to call [updateConfig](https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/pkg/bootstrapper/kubelet.go#L121) ?

Reply:
sorry you are right, read it too quick and thought that was refreshing the manager